src/services/monitoring.ts

```diff
@@ -10,422 +26,530 @@ const {
-  outcome: string;
-  findingStatus: string;
-  specialists: {
-    name: string;
-    roles: string[];
-  }[];
-  objectives: ITTAByReviewObjective[];
-}
+  return objectives.map((objective) => {
+    let findingIds = [];
+    let reviewNames = [];
+    let endDate = null;
+    const grants = [];
+    const activityReports = [];
+    const specialists = [];
+    const topics = [];
 
-interface ITTAByReviewFinding {
-  citation: string;
-  status: string;
-  type: string;
-  category: string;
-  correctionDeadline: string;
-  objectives: ITTAByReviewObjective[];
-}
+    const { activityReportObjectives } = objective;
+
+    activityReportObjectives.forEach((activityReportObjective) => {
+      const {
+        activityReportObjectiveCitations,
+        activityReport,
+        activityReportObjectiveTopics,
+      } = activityReportObjective;
+      const { activityReportCollaborators, author } = activityReport;
+
+      activityReportObjectiveCitations.forEach((citation) => {
+        findingIds = findingIds.concat(citation.findingIds);
+        grants.push(citation.grantNumber);
+        reviewNames = reviewNames.concat(citation.reviewNames);
+      });
+
+      specialists.push({ name: author.fullName, roles: author.roles.map((role) => role.name) });
+      activityReportCollaborators.forEach((collaborator) => {
+        specialists.push({ name: collaborator.user.fullName, roles: collaborator.user.roles.map((role) => role.name) });
+      });
+
+      activityReportObjectiveTopics.forEach((topic) => {
+        topics.push(topic.topic.name);
+      });
 
-interface ITTAByReviewResponse {
-  name: string;
-  reviewType: string;
-  reviewReceived: string;
-  findings: ITTAByReviewFinding[];
-  grants: string[];
-  outcome: string;
-  lastTTADate: string | null;
-  specialists: {
-    name: string;
-    roles: string[];
-  }[];
+      activityReports.push({
+        id: activityReport.id,
+        displayId: activityReport.displayId,
+      });
+
+      if (!endDate || moment(activityReport.endDate).isAfter(endDate)) {
+        endDate = moment(activityReport.endDate);
+      }
+    });
+
+    return {
+      findingIds,
+      grantNumber: grants.join('\n'),
+      reviewNames,
+      title: objective.title,
+      activityReports,
+      endDate: endDate ? endDate.format('MM/DD/YYYY') : null,
+      topics: uniq(topics),
+      status: objective.status,
+      specialists,
+    };
+  });
 }
 
-interface ITTAByCitationResponse {
-  citationNumber: string;
-  findingType: string;
-  status: string;
-  category: string;
-  grantNumbers: string[];
-  lastTTADate: string | null;
-  reviews: ITTAByCitationReview[];
+async function extractExternalData(recipientId: number, regionId: number) {
+  const grantNumbers = await grantNumbersByRecipientAndRegion(recipientId, regionId) as string[];
+  const citationsOnActivityReports = await aroCitationsByGrantNumbers(grantNumbers);
+
+  const monitoringReviewGrantees = await MonitoringReviewGrantee.findAll({
```

Comment:
Is there any concern here for taking into account the date with this or the aroCitationsByGrantNumbers()?